packages/@aws-cdk/core/lib/private/uniqueid.ts

```diff
@@ -45,7 +46,7 @@ export function makeUniqueId(components: string[]) {
   // top-level resources will simply use the `name` as-is in order to support
   // transparent migration of cloudformation templates to the CDK without the
   // need to rename all resources.
-  if (components.length === 1) {
+  if (components.length === 1 && components[0].length < MAX_ID_LEN) {
```

Comment:
We need to check the max length after removing nonalphanums, no?

Reply:
Good catch. Otherwise it's can potentially be a breaking change. ON IT.

Reply:
The right approach would have been to use the hash mode if there are any non-alpha (because otherwise we can get a conflict). But this will be a breaking change.

Reply:
https://github.com/aws/aws-cdk/issues/6421